src/briefcase/integrations/docker.py

```diff
@@ -378,9 +381,377 @@ def check_output(self, args: list[SubprocessArgT], image_tag: str) -> str:
+                "The DISPLAY environment variable must be set to run an app in Docker"
+            )
+
+        # Create a TCP proxy for a spoofed X display
+        proxy_popen, proxy_display_num = self._x11_tcp_proxy(DISPLAY)
+
+        # Define the xauth database files for the spoofed display
+        xauth_file_path = self._x11_proxy_display_xauth_file_path(proxy_display_num)
+        docker_xauth_file_path = PurePosixPath("/tmp") / xauth_file_path.name
+        try:
+            # Create an x auth file for the spoofed display based on current x auth
+            self._x11_write_xauth_file(DISPLAY, xauth_file_path, proxy_display_num)
+
+            # Augment the subprocess kwargs so the container uses the spoofed display
+            subprocess_kwargs.setdefault("mounts", []).append(
+                (xauth_file_path, docker_xauth_file_path)
+            )
+            subprocess_kwargs.setdefault("env", {}).update(
+                {
+                    # Use host.docker.internal as the hostname for the display which
+                    # will be dynamically mapped to the network interface for the host
+                    "DISPLAY": f"host.docker.internal:{proxy_display_num}",
+                    # Tell X clients which xauth to use to connect to the display
+                    "XAUTHORITY": docker_xauth_file_path,
+                }
+            )
+            subprocess_kwargs.setdefault("add_hosts", []).extend(
+                [
+                    # This adds a mapping to the container's /etc/hosts for
+                    # `host.docker.internal` to a network address for the host.
+                    # The keyword `host.docker.internal` is already always defined in
+                    # Docker Desktop via the DNS configured within its Linux VM.
+                    # However, for Docker Engine, DNS is managed by the Linux host;
+                    # therefore, this must be defined each time the container is started.
+                    # The keyword `host-gateway` is translated by the Docker server to
+                    # a network address for the host that is reachable from within the
+                    # container.
+                    ("host.docker.internal", "host-gateway"),
+                ]
+            )
+            yield subprocess_kwargs
+        finally:
+            self.tools.subprocess.cleanup("X display proxy", proxy_popen)
+            xauth_file_path.unlink(missing_ok=True)
+
+    def _x11_tcp_proxy(self, DISPLAY: str) -> tuple[subprocess.Popen, int]:
+        """Starts a TCP proxy as a spoofed X display.
+
+        The TCP server's port is chosen as 6000 + the display number as defined in the
+        X11 standard. In this way, when X clients attempt to make a connection for the
+        chosen display number, this TCP port will be automatically evaluated by the
+        client for consideration as an X display.
+
+        The proxy is binded to 0.0.0.0 and as such will be reachable on all network
```

Comment:
```suggestion
        The proxy is bound to 0.0.0.0 and as such will be reachable on all network
```

Reply:
This one's kinda funny....my IDE has had this squiggly-underlined since I wrote it. It's mostly a personal disagreement I have with current rules of English. To my benefit of my viewpoint, though, there is at least one relevant example: hung vs hanged...although, a bit more graphic. Nonetheless, I'll relent.

Reply:
Don't get me started on the US need to use "drug" as a the past participle of drag. :-)